ebill_paynet/models/paynet_service.py

```diff
@@ -0,0 +1,223 @@
+    )
+    invoice_message_ids = fields.One2many(
+        comodel_name='paynet.invoice.message',
+        inverse_name='service_id',
+        string='Invoice Messages',
+        readonly=True,
+    )
+    ebill_payment_contract_ids = fields.One2many(
+        comodel_name='ebill.payment.contract',
+        inverse_name='paynet_service_id',
+        string='Contracts',
+        readonly=True,
+    )
+    active = fields.Boolean(default=True)
+
+    @api.depends('use_test_service')
+    def _compute_auth_from_env(self):
+        for record in self:
+            if not record.use_test_service:
+                prefix = 'PAYNET'
+            else:
+                prefix = 'PAYNET_TEST'
+            record.username = os.getenv(prefix + '_USERID')
+            record.password = os.getenv(prefix + '_PASSWORD')
+
+    @api.depends('use_test_service')
+    def _compute_url(self):
+        for record in self:
+            if record.use_test_service:
+                record.url = SYSTEM_TEST_URL
+            else:
+                record.url = SYSTEM_PROD_URL
+
+    @api.multi
+    def take_shipment(self, content):
+        """Send a shipment via DWS to the Paynet System
+
+        Return value is the shipment id
+        """
+        self.ensure_one()
+        dws = PayNetDWS(self.url, self.use_test_service)
+        content = content.encode('utf-8')
+        res = dws.service.takeShipment(
+            Authorization=dws.authorization(self.username, self.password),
+            # ProcessingDate  : Preferred processing date,
+            #                   if not provided, processed asap
+            # ShipmentPriority: Value between 1 and 9 (default is 5)
+            Content=content,
+        )
+        return res
+
+    @api.multi
+    def get_shipment_list(self):
+        """Get a list of shipments present on the DWS."""
+        self.ensure_one()
+        dws = PayNetDWS(self.url, self.use_test_service)
+        res = dws.service.getShipmentList(
+            Authorization=dws.authorization(self.username, self.password),
+            # fromEntry     : Position number as of which shipments should be
+            #                 retrieved (default is 1)
+            # maxEntries    : Max number of shimpment listed (default is 100)
+            # FromDate      :
+            # ToDate        :
+            ShipmentStates=PENDING_STATES,
+            # FromShipmentPriority:
+            # ToShipmentPriority:
+        )
+        return res
+
+    @api.multi
+    def get_shipment_content(self, shipment_id):
+        """ """
+        self.ensure_one()
+        dws = PayNetDWS(self.url, self.use_test_service)
+        try:
+            res = dws.client.service.getShipmentContent(
+                Authorization=dws.authorization(self.username, self.password),
+                ShipmentID=shipment_id,
+            )
+        except Fault as e:
+            dws.handle_fault(e)
+            # raise
```

Comment:
Useless line?

Reply:
It was a reminder, should it be raised and handled further up ?